sonic_platform_base/sonic_storage/ssd.py

```diff
@@ -427,5 +435,11 @@ def get_vendor_output(self):
         """
         return self.vendor_ssd_info
 
-    def parse_id_number(self, id):
-        return self._parse_re('{}\s*(.+?)\n'.format(id), self.ssd_info)
+    def parse_id_number(self, id, buffer):
+        if buffer:
+            buffer_lines = buffer.split('\n')
+            for line in buffer_lines:
+                if line.startswith(str(id)):
```

Comment:
Should we strip the whitespace in the front and back of the line?  For example I see the below output. Will it be parsed for id =9 or 10?

```
  9 Power_On_Hours          0x0002   036   000   000    Old_age   Always       -       54308
 10 Spin_Retry_Count        0x0000   000   000   000    Old_age   Offline      -       0
 12 Power_Cycle_Count       0x0002   143   000   000    Old_age   Always       -       9615
163 Total_Bad_Block_Count   0x0000   000   000   000    Old_age   Offline      -       13
```

Reply:
Oh I see what you mean -- I'll make that change real quick.

Reply:
Changed in latest